### PR TITLE
Add `Key128bit` to wrap `sgx_key_128bit_t`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #91** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/91" target="_blank">Graphite</a> 
    * **PR #86** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/86" target="_blank">Graphite</a> 
      * **PR #92** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/92" target="_blank">Graphite</a> 
        * **PR #103** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/103" target="_blank">Graphite</a> 
          * **PR #110** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/110" target="_blank">Graphite</a> 
            * **PR #111** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/111" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/111?utm_source=stack-comment).

File: core/types/src/key_request.rs
Comment (nick-mobilecoin) [inline core/types/src/key_request.rs:169]: This is in `key_request.rs` it may be that the file should be renamed to `key.rs` or similar
The request gets a `Key128bit` so the two are pretty closley tied, but i could also see that the request and the key each get their own module.

File: core/types/src/key_request.rs
Comment (jcape) [inline core/types/src/key_request.rs:180]: Is this constant useful outside this file? Can this be made private?

If it's going to be used outside, we should (ugh) make this into something like a `FixedSize` trait or something...

**Review by jcape — APPROVED** [review]
LGTM, but I question whether the `SIZE` constant needs to be public or not.

File: core/types/src/key_request.rs
Comment (nick-mobilecoin) [reply core/types/src/key_request.rs]: Removed it.  Along with being able to derive `Default` it was over exposing API members thinking users might need to know how to construct a key.  They should get the default and then pass that as a reference to `sgx_get_key()`

**Review by awygle — APPROVED** [review]
This looks good. I prefer to have the size constants accessible, personally, but I'm not going to be precious about it.

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/111?utm_source=stack-comment).


Job Reference: C6ZyZ0Ggec8V0ksW3mYN

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/111?utm_source=stack-comment).


Job Reference: ANN5tbEFcYrOel5JRpoC

**Pull request merged**
